datasource/src/main/scala/quasar/physical/blobstore/azure/AzureDatasource.scala

```diff
@@ -17,17 +17,23 @@
 package quasar.physical.blobstore.azure
 
 import quasar.api.datasource.DatasourceType
-import quasar.blobstore.azure.AzureBlobstore
+import quasar.blobstore.azure.{Azure, AzureBlobstore, AzureConfig, MaxQueueSize}
 import quasar.connector.MonadResourceErr
 import quasar.physical.blobstore.BlobstoreDatasource
 
 import cats.Applicative
+import cats.effect.ConcurrentEffect
+import cats.syntax.functor._
 import eu.timepit.refined.auto._
 import fs2.RaiseThrowable
 
 class AzureDatasource[F[_]: Applicative: MonadResourceErr: RaiseThrowable](azureBlobstore: AzureBlobstore[F])
   extends BlobstoreDatasource[F](AzureDatasource.dsType, azureBlobstore)
 
 object AzureDatasource {
-  val dsType = DatasourceType("azure", 1L)
+  val dsType: DatasourceType = DatasourceType("azure", 1L)
+
+  def mk[F[_]: ConcurrentEffect: MonadResourceErr](cfg: AzureConfig): F[AzureDatasource[F]] =
```

Comment:
No strong opinions on this, but my personal preference is to use `apply` only when both of these conditions hold:

1) the type that's returned exactly matches the name of the class (i.e. not `F[MyClass]`)
2) there are no extra heavy dependencies pulled in compared to the class constructor

I didn't use `apply` here mainly because of 1)